Python/bytecodes.c

```diff
@@ -971,15 +971,14 @@ dummy_func(
             PyObject *sub = PyStackRef_AsPyObjectBorrow(sub_st);
             PyObject *dict = PyStackRef_AsPyObjectBorrow(dict_st);
 
-            assert(PyDict_CheckExact(dict));
+            assert(PyDict_Check(dict));
```

Comment:
I think you also need to change `_GUARD_NOS_DICT` to `PyDict_Check` instead of `PyDict_CheckExact` for this to work but I am not sure.

Reply:
It does look like it from the code, but I don't see more misses. 

<img width="961" alt="image" src="https://github.com/user-attachments/assets/503e2b6a-f82d-4d6c-9868-526534f36b0c" />

Reply:
Changing `_GUARD_NOS_DICT` would break `STORE_SUBSCR_DICT`